CMakeLists.txt

```diff
@@ -1,4 +1,5 @@
 cmake_minimum_required(VERSION 3.10)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
What if I switch to C++20 and QtOlm happens to use things dropped from C++20? Also, I'm going to add QtKeychain to the list of the library dependencies; this change will mean that if QtKeychain is not compliant with whichever  CXX_STANDARD I use I'll have to patch it or wait for the upstream.

In other words, I'd prefer to not impose C++ level requirements on other libraries - rather, I expect the library to expose their C++ standard through respective PROPERTIES.

Reply:
I'm not sure... The CI failed if CMAKE_CXX_STANDARD is not specified.

Reply:
Am I missing something?

Reply:
That was broken even before your PR - qmc-example doesn't have CXX_STANDARD set, and the library's property is not exposed.

Just found that the right way of specifying the C++ standard seems to be via `target_compile_features()`: https://cmake.org/cmake/help/v3.10/manual/cmake-compile-features.7.html#requiring-language-standards.

Reply:
Ah, ok